eng/pipelines/templates/steps/java-publishing.yml

```diff
@@ -74,3 +75,19 @@ steps:
           -RepositoryPassword $(azuresdk-sonatype-password)
           -GPGExecutablePath ${{ parameters.BuildToolsPath }}/tools/gpg/bin/gpg.exe
           -InformationAction Continue
+
+  - ${{if eq(parameters.Target, 'MavenCentralPartner')}}:
```

Comment:
Not sure about this. When used in our unified pipelines we pass along ArtifactId and GroupId filters, but that isn't necessarily in the partner release pipeline because we publish whatever is in the directory. This means that there are two quite distinct invocations of the template.

Reply:
If possible we should only have MavenCentral and have the partner release pipeline pass in the correct wildcards to publish all.

Reply:
OK, the way I approached this was to collapse both invocations in under MavenCentral but to use one invocation or the other based on the artifact ID or group ID parameters. Once inside the script I largely avoid wildcards after I've done the parse of POM files to avoid accidentally picking up artifacts that I don't want.